stock_valuation_fifo_lot/models/product.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+from odoo.tools import float_is_zero
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_all_candidates(self, company):
+        all_candidates = (
+            self.env["stock.valuation.layer"]
+            .sudo()
+            .search(
+                [
+                    ("product_id", "=", self.id),
+                    ("remaining_qty", ">", 0),
+                    ("company_id", "=", company.id),
+                ]
+            )
+            .sorted(lambda l: l.lot_ids.create_date)
+        )
```

Comment:
(BTW, we can resolve singleton issue here by combining `min` and `mapped`.)
```suggestion
        )
        if sort_by == 'lot_create_date':
            sort_by = lambda l: min(l.lot_ids.mapped('create_date'))
        if sort_by is not None:
            all_candidates = all_candidates.sorted(sort_by)
```